lib/uiautomator2.js

```diff
@@ -129,12 +129,17 @@ class UiAutomator2Server {
         if (!isAmServiceAvailable) {
           // https://travis-ci.org/appium/appium-uiautomator2-driver/jobs/478514097#L4806
           amError = null;
+          let stdout;
           try {
-            await this.adb.shell(['am']);
-            log.debug('Activity manager service is available');
-            isAmServiceAvailable = true;
+            stdout = await this.adb.shell(['am stack list > /dev/null;echo $?']);
           } catch (e) {
             amError = e;
+            stdout = e.stdout;
+          }
+          if (parseInt(stdout, 10) === 0) {
```

Comment:
This could use a comment, as it is not obvious what it is doing.

Reply:
yep, added a comment